tardisbase/testing/regression_comparison/visualize_files.py

```diff
@@ -0,0 +1,245 @@
+        older_files = self.get_h5_files_in_commit(older_commit)
+        newer_files = self.get_h5_files_in_commit(newer_commit)
+        all_files = older_files | newer_files
+
+        changes = {}
+        for file_path in all_files:
+            in_older = file_path in older_files
+            in_newer = file_path in newer_files
+
+            if not in_older and not in_newer:
+                changes[file_path] = "−"
+            elif not in_older and in_newer:
+                changes[file_path] = "A"
+            elif in_older and not in_newer:
+                changes[file_path] = "D"
+            elif self.is_file_modified(file_path, older_commit, newer_commit):
+                changes[file_path] = "M"
+            else:
+                changes[file_path] = "•"
+
+        return changes
+
+
+    def create_file_data_row(self, file_path):
+        """
+        Create a data row for the file change matrix.
+
+        Parameters
+        ----------
+        file_path : str
+            File path for the row.
+
+        Returns
+        -------
+        dict of str : str
+            Row data with file path and change symbols.
+        """
+        row = {'Files': file_path}
+        for transition_key in self.transition_columns:
+            change_type = self.file_transitions.get(transition_key, {}).get(file_path, "−")
+            row[transition_key] = change_type
+        return row
+
+
+    def print_dataframe_matrix(self):
+        """
+        Print the file change matrix as a DataFrame.
+        """
+        all_data = [self.create_file_data_row(file_path)
+                   for file_path in sorted(self.all_files)]
+
+        if all_data:
+            df = pd.DataFrame(all_data)
+            print(f"\nFile Changes Matrix ({len(self.all_files)} files):")
+            print("=" * 60)
+            print("Legend: A = Added  |  D = Deleted  |  M = Modified  |  • = Unchanged  |  − = Not-Present")
+            print()
+            display(df)
+        else:
+            print("\nNo files found across the analyzed transitions.")
+
+
+    def print_commit_info(self):
+        """
+        Print commit information table.
+        """
+        commit_data = []
+
+        commit_type = ("Generated Commits from TARDIS" if self.tardis_commits
+                      else "Direct Regression Data Commits")
+
+        for i, commit_hash in enumerate(self.commits):
+            commit = self.repo.commit(commit_hash)
+            description = f"Regression data for --{commit.message.strip().split('\n')[0][:60]}"
+
+            commit_data.append({
+                'Commit #': i + 1,
+                'Regression Hash': commit_hash[:8],
+                'Description': description,
+                'Date': commit.committed_datetime.strftime('%Y-%m-%d %H:%M')
+            })
+
+        df = pd.DataFrame(commit_data)
+        print(f"\nCOMMIT INFORMATION ({len(self.commits)} commits) - {commit_type}:")
+        print("=" * 80)
+        display(df)
```

Comment:
dont display- just return, running these in a separate cell will automatically display them